jtsx.el

```diff
@@ -861,29 +861,79 @@ ELEMENT-NAME is the name of the new wrapping element."
-                 (goto-char end-pos)))
-    ;; Starting Emacs 30, treesit set its own function, which has some issues.
-    ;; Bug report: https://debbugs.gnu.org/cgi/bugreport.cgi?bug=66988
-    ;; Use the default one instead.
-    (let ((forward-sexp-function nil))
-      (forward-sexp n))))
+See `forward-sexp' documentation for informations about ARG and
+INTERACTIVE arguments.
+Note that ARG values other than 1 and -1 are ingnored inside JSX context."
+  (interactive "^p\nd")
+  (if (jtsx-jsx-context-p)
+    (cond
+     ((and (number-or-marker-p arg) (< arg 0))
+      (when-let* ((node (treesit-node-at (point)))
+                  (enclosing-node
+                   (jtsx-enclosing-jsx-node node jtsx-jsx-hs-root-keys))
+                  (start-pos (treesit-node-start enclosing-node)))
+        (goto-char start-pos)))
+     (t (when-let* ((node (treesit-node-at (point)))
+                    (enclosing-node
+                     (jtsx-enclosing-jsx-node node jtsx-jsx-hs-root-keys))
+                    (end-pos (treesit-node-end enclosing-node)))
+          (goto-char end-pos))))
+
+    (if (or
+         ;; Starting Emacs 30, treesit set its own function, which has
+         ;; some issues. Bug report:
+         ;; https://debbugs.gnu.org/cgi/bugreport.cgi?bug=66988 Use the
+         ;; default one instead.
+         (>= 30 emacs-major-version)
+
+         ;; Prevent recursive call
+         (eq forward-sexp-function #'jtsx-hs-forward-sexp))
+        (let ((forward-sexp-function nil))
+          (forward-sexp arg interactive))
+      (forward-sexp arg interactive))))
+
+(defun jtsx-backward-up-list
+    (&optional arg escape-strings no-syntax-crossing should-push-mark)
+  "Adding `backward-up-list' support when inside JSX block.
+If SHOULD-PUSH-MARK is non-nil (as it is interactively), call
+`push-mark' before moving point to another position.
+Note that ARG is ignored inside JSX context.  For ESCAPE-STRINGS and
+NO-SYNTAX-CROSSING, Please see `backward-up-list'."
+  (interactive "^p\nd\nd\nd")
+  (if (jtsx-jsx-context-p)
+      (when-let* ((node (treesit-node-at (point)))
+                  (enclosing-node
+                   (jtsx-enclosing-jsx-node node '("jsx_element"
+                                                   "jsx_self_closing_element")))
+                  (parent-node (treesit-node-parent enclosing-node))
+                  (start-pos (treesit-node-start parent-node)))
+        (when should-push-mark (push-mark nil t nil))
+        (goto-char start-pos))
+    (let ((orig-pos (point)))
+      (condition-case nil
+          (progn
+            (backward-up-list arg escape-strings no-syntax-crossing)
+            (when should-push-mark (push-mark orig-pos t nil)))
+        ((scan-error user-error)
+         (goto-char orig-pos))))))
+
+(defun jtsx-hs-looking-at-block-start-p ()
+  "Return non-nil if the point is at the block start."
+  (if (jtsx-jsx-context-p)
+      (looking-at hs-block-start-regexp)
```

Comment:
After looking deeper at this piece of code, my understanding is that it checks whether inside a comment or not, probably to differentiate uncommented and commented code when folding ?

So this leads me to think that maybe we should do an equivalent check in jsx context using a `tree-sitter` way (or  fix `syntax-table` for jsx code ?).

But since jsx comments folding seems to only partially work, this first fix can be used to repair `Hideshow` folding as soon as possible (completly broken for now) with very limited side effects.